src/otaclient/ota_core.py

```diff
@@ -1330,7 +1323,7 @@ def update(self, request: UpdateRequestV2) -> None:
                 upper_otaproxy=self.proxy,
                 status_report_queue=self._status_report_queue,
                 session_id=new_session_id,
-                metrics=replace(self._metrics, session_id=new_session_id),
+                metrics=self._metrics,
```

Comment:
Changed to use a single instance instead of creating a new instance per session because:

- We want to modify the metrics field later in the `_on_failure` function without having to pass the metrics object around.

- The object includes a dedicated shared memory resource, making it simpler and safer to manage as a single shared object rather than having multiple objects sharing the same memory.